eng/pipelines/templates/jobs/vmr-build.yml

```diff
@@ -87,8 +90,21 @@ jobs:
     - name: sourcesPath
       value: ${{ parameters.vmrPath }}
 
+  templateContext:
+    outputs:
+    - output: pipelineArtifact
+      displayName: 'Publish BuildLogs'
+      condition: succeededOrFailed()
+      targetPath: '$(Build.StagingDirectory)/BuildLogs'
+      artifactName: $(Agent.JobName)_BuildLogs_Attempt$(System.JobAttempt)
+    - output: pipelineArtifact
+      displayName: 'Publish Artifacts'
+      condition: succeededOrFailed()
+      targetPath: '${{ variables.sourcesPath }}/artifacts/${{ parameters.architecture }}/Release/'
+      artifactName: $(Agent.JobName)_Artifacts
+
   steps:
-  - template: ../steps/vmr-prepare.yml
+  - template: ../steps/vmr-prepare.yml@self
```

Comment:
FYI I found out that the `@self` is not really needed in any templates that are not top-level with `extends`. I.e. lines 71 and 73 don't have this and it still works.